tests/test_aws.py

```diff
@@ -1,20 +1,35 @@
+# Tests requiring S3 credentials.
+# Collected here to make them easier to skip/xfail.
+
 import logging
-import os
 import sys
 
+from packaging.version import parse
 import pytest
 
 import rasterio
 from rasterio.aws import Session
+from rasterio.rio.main import main_group
+
+
+# Custom markers.
+mingdalversion = pytest.mark.skipif(
+    parse(rasterio.__gdal_version__).base_version < parse('2.1.0'),
+    reason="S3 raster access requires GDAL 2.1")
+
+credentials = pytest.mark.skipif(
+    not(Session()._creds), reason="S3 raster access requires credentials")
```

Comment:
@perrygeo check out these custom markers. Here's where the new `__gdal_version__` gets used.